### PR TITLE
feat: added observability handler id as span attribute

### LINKED ISSUE #180: Add ObservabilityHandler identifier to front-end spans attributes
An `ObservabilityHandler` instance is created and associated to every UI instance, so its identifier could be used to group front-end traces that come from the same browser tab.

`ClientInstumentation` can add the `ObservabilityHandler` identifier as an attribute to all the processed spans.

This should also work fine with `@PreserveOnRefresh`, since the UI instance will not change on page reload and the  `ObservabilityHandler` will not be recreated.

### PR DESCRIPTION
**Pull request by mcollovati**
## Description

Having  observability handler id as span attribute allow to filter traces produces by the same browser tab.
Also replaces span kind to be CLIENT for front-end spans.

Closes #180

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: observability-kit-agent/src/test/java/com/vaadin/extension/instrumentation/client/ClientInstrumentationTest.java [hunk 1]
```diff
@@ -52,11 +59,21 @@ public void handleRequest_createsParentAndChildSpans() {
...
             SpanData childSpan = getExportedSpan(0);
             assertEquals(parentSpan.getSpanId(), childSpan.getParentSpanId());
+            assertEquals(observabilityClientId,
+                    parentSpan.getAttributes().get(AttributeKey.stringKey(
+                            ClientInstrumentation.MethodAdvice.FRONTEND_ID)),
+                    "Missing or invalid observability client identifier on parent span attributes");
+            assertEquals(observabilityClientId,
+                    childSpan.getAttributes().get(AttributeKey.stringKey(
+                            ClientInstrumentation.MethodAdvice.FRONTEND_ID)),
+                    "Missing or invalid observability client identifier on parent span attributes");
```
[the remark this answers is not in the document]
Comment (mcollovati) [reply observability-kit-agent/src/test/java/com/vaadin/extension/instrumentation/client/ClientInstrumentationTest.java:76]: Good catch! Thanks

[the remark this answers is not in the document]
Comment (mcollovati) [reply observability-kit-agent/src/test/java/com/vaadin/extension/instrumentation/client/ClientInstrumentationTest.java]: done

File: observability-kit-agent/src/main/java/com/vaadin/extension/instrumentation/client/ClientInstrumentation.java [hunk 2]
```diff
@@ -46,8 +47,12 @@ public void transform(TypeTransformer transformer) {
     }
 
     public static class MethodAdvice {
+
+        static final String FRONTEND_ID = "vaadin.front-end.id";
```
Comment (heruan) [inline observability-kit-agent/src/main/java/com/vaadin/extension/instrumentation/client/ClientInstrumentation.java:51]: Nit: elsewhere we spell `frontend` and `backend` without the hyphen.
```suggestion
        static final String FRONTEND_ID = "vaadin.frontend.id";
```

File: observability-kit-test/observability-kit-agent-it/src/test/java/com/vaadin/observability/test/MainViewIT.java [hunk 3]
```diff
@@ -77,6 +78,10 @@ public void verifyFrontendTraces() {
             var spans = extractSpansFromRequests(requests);
             assertThat(spans).extracting(Span::getName)
                     .contains("Client: documentLoad", "Client: documentFetch");
+            assertThat(spans)
+                    .filteredOn(span -> span.getKind() == Span.SpanKind.SPAN_KIND_CLIENT)
+                    .allSatisfy(span -> assertThat(span.getAttributesList())
+                            .extracting(KeyValue::getKey).contains("vaadin.front-end.id"));
```
Comment (heruan) [inline observability-kit-test/observability-kit-agent-it/src/test/java/com/vaadin/observability/test/MainViewIT.java:84]: Same here.
```suggestion
                            .extracting(KeyValue::getKey).contains("vaadin.frontend.id"));
```

**Review by heruan — CHANGES_REQUESTED** [review]

**Review by heruan — APPROVED** [review]

**Pull request merged**